### ISSUE TITLE
PHP notice due to unsaved options

### ISSUE THREAD

**Issue by boonebgorges** [semperfiwebdesign/all-in-one-seo-pack#2805]
Please fill out this form as completely as possible so that we can help you more quickly.

- [x] I understand that this is a development repository and not a support forum. This is a bug report and is not a support request.
 - [x] I've followed the [troubleshooting steps](https://semperplugins.com/documentation/how-to-troubleshoot-issues-with-all-in-one-seo-pack)  to determine if there's a theme/plugin conflict. 

**Provide the relevant URL(s) here**

**Describe the bug**
Before plugin settings are initially saved, plugin throws PHP notices because of `array_key_exists()` checks in `check_recently_added_modules()`. See https://github.com/semperfiwebdesign/all-in-one-seo-pack/blob/189d80c5d37694b73175954ea221ff292ff076eb/aioseop_class.php#L5701

The best solution may be to have a central function for fetching the plugin options, a function which would ensure that proper defaults are in place. An easier fix is simply to change the line in question to:

```php
$options = get_option( 'aioseop_options', array() );
```

to ensure that the return value will be in an array.

Thanks!

**Comment by michaeltorbert** [comment]
@boonebgorges Thanks for the report and input. 
The best way to make sure your issue is resolved and included in the next release is to submit a PR. Feel free to submit one if you like, which will have the added bonus of listing you in the credits of All in One SEO Pack. ;)

**Comment by michaeltorbert** [comment]
Although I'm not sure where check_recently_added_modules() is... Is that a different plugin you're using or am I overlooking something?

**Comment by michaeltorbert** [comment]
@boonebgorges We'll also need steps to reproduce.

**Comment by wpsmort** [comment]
@boonebgorges Can you provide us with the steps to reproduce this PHP notice?

**Comment by boonebgorges** [comment]
Not exactly. My guess is that it's a race condition that is hard to trigger consistently, as it requires two nearly-simultaneous requests of a specific type. See "Steps to Reproduce" at https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/2807
